### ISSUE TITLE
oninit is not running for nested components

### ISSUE THREAD

**Issue by SamuelTilly** [StephanHoyer/mithril-query#54]
When using the rewrite branch mq is not running component `oninit` function for nested components.

My simple guess is that in https://github.com/StephanHoyer/mithril-query/blob/rewrite/index.js#L106
`states[treePath]` is already defined and therefore `oninit` is never triggered.

``` javascript
test('component', function(t) {
    var result = {
        oninit: 0,
        view: 0,
    };

    var component = {
        oninit: function() {
            result.oninit++;
        },
        view: function(vnode) {
            result.view++;
            return m("div.test", vnode.children);
        },
    };

    mq(
        m(component, [
            m(component, [
                m(component), // <-- oninit is not running for this component.
            ]),
        ])
    );

    t.equal(result.oninit, result.view, 
        'oninit should run before every view');

    t.end();
});
```

**Comment by StephanHoyer** [comment]
I currently haven't the time to fix this. PR would be highly appreciated.

**Comment by StephanHoyer** [comment]
fixed with #65
